iree/compiler/Codegen/SPIRV/SPIRVTile.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===- SPIRVTile.cpp ------------------------------------------------------===//
+//
+// This pass tiles and Linalg ops with tensor semantics to invocations.
+//
+//===----------------------------------------------------------------------===//
+
+#include "iree/compiler/Codegen/Dialect/LoweringConfig.h"
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "iree/compiler/Codegen/SPIRV/Utils.h"
+#include "iree/compiler/Codegen/Utils/MarkerUtils.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/Linalg/IR/LinalgOps.h"
+#include "mlir/Dialect/SCF/SCF.h"
+#include "mlir/Dialect/SCF/Transforms.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-spirv-tile"
+
+namespace mlir {
+namespace iree_compiler {
+
+//===----------------------------------------------------------------------===//
+// Tiling patterns
+//===----------------------------------------------------------------------===//
+
+/// Populates `patterns` with patterns that tiles convolution/matmul ops with
+/// markers.
+static void populateTilingReductionPatterns(RewritePatternSet &patterns) {
+  MLIRContext *context = patterns.getContext();
+  auto getTileSizeFn = [&](OpBuilder &builder, Operation *op) {
+    return getTileSizes(builder, op, 2);
+  };
+  auto tilingOptions = linalg::LinalgTilingOptions()
+                           .setLoopType(linalg::LinalgTilingLoopType::Loops)
+                           .setTileSizeComputationFunction(getTileSizeFn);
+  auto marker = Identifier::get(getTileReductionMarker(), context);
+  auto filter = linalg::LinalgTransformationFilter({marker});
```

Comment:
This broke our gcc build (tested on post-submit). Logs here: https://buildkite.com/iree/iree-build-configurations/builds/1496#91c054cd-df7a-4758-84bd-91a621668581

error message:
```
[2021-12-02T19:07:52Z] ../iree/compiler/Codegen/SPIRV/SPIRVTile.cpp: In function 'void mlir::iree_compiler::populateTilingReductionPatterns(mlir::RewritePatternSet&)':
[2021-12-02T19:07:52Z] ../iree/compiler/Codegen/SPIRV/SPIRVTile.cpp:44:60: error: call of overloaded 'LinalgTransformationFilter(<brace-enclosed initializer list>)' is ambiguous
[2021-12-02T19:07:52Z]    44 |   auto filter = linalg::LinalgTransformationFilter({marker});
[2021-12-02T19:07:52Z]       |                                                            ^
[2021-12-02T19:07:52Z] In file included from ../iree/compiler/Codegen/Passes.h:15,
[2021-12-02T19:07:52Z]                  from ../iree/compiler/Codegen/SPIRV/SPIRVTile.cpp:15:
[2021-12-02T19:07:52Z] ../third_party/llvm-project/llvm/../mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h:436:3: note: candidate: 'mlir::linalg::LinalgTransformationFilter::LinalgTransformationFilter(const mlir::linalg::LinalgTransformationFilter&)'
[2021-12-02T19:07:52Z]   436 |   LinalgTransformationFilter(const LinalgTransformationFilter &) = default;
[2021-12-02T19:07:52Z]       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~
[2021-12-02T19:07:52Z] ../third_party/llvm-project/llvm/../mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h:435:3: note: candidate: 'mlir::linalg::LinalgTransformationFilter::LinalgTransformationFilter(mlir::linalg::LinalgTransformationFilter&&)'
[2021-12-02T19:07:52Z]   435 |   LinalgTransformationFilter(LinalgTransformationFilter &&) = default;
[2021-12-02T19:07:52Z]       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~
[2021-12-02T19:07:52Z] ../third_party/llvm-project/llvm/../mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h:427:12: note: candidate: 'mlir::linalg::LinalgTransformationFilter::LinalgTransformationFilter(llvm::ArrayRef<mlir::StringAttr>, llvm::Optional<mlir::StringAttr>)'
[2021-12-02T19:07:52Z]   427 |   explicit LinalgTransformationFilter(
[2021-12-02T19:07:52Z]       |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Sent a fix: https://github.com/google/iree/pull/7802

Reply:
Ah, sorry about this. Thanks for the fix!!